### PR TITLE
introducing options

### PR DESCRIPTION
**Pull request by adnasa**
As discussed, we should pass in `MAX_RESULTS` as a `size` option into the `execute` function.

so now `execute` has the following signature

```js
execute (<String> query, <Object> options)
```

I also renamed `q` to `query` as well in the core plugins.

### CODE REVIEW

**Comment by vutran** [comment]
Can you rebase with latest changes in `develop`?

**Comment by adnasa** [comment]
@vutran a240ba7

File: app/main/plugins.js [hunk 1]
```diff
@@ -228,17 +230,15 @@ exports.queryResults = (plugin, args) => new Promise(resolve => {
...
-      if (typeof m.execute === 'function') {
-        output = m.execute(q);
-      } else {
-        output = m.execute;
-      }
-      let items = [];
+      // eslint-disable-next-line global-require
+      const pluginObj = require(plugin.path);
+      const output = (typeof pluginObj.execute === 'function')
+        ? pluginObj.execute(query, { size: MAX_RESULTS })
```
Comment (adnasa) [inline app/main/plugins.js]: Now `giphy` and other plugins can set a limit parameter on their respective apis 🎉

Comment (adnasa) [inline app/main/plugins.js]: Should probably check to update the documentation accordingly

**Comment by adnasa** [comment]
don't merge yet.. doc udpate incoming

**Comment by adnasa** [comment]
okay, merge ready

**Comment by adnasa** [comment]
okay, I found one more thing 😄

File: resources/plugins/bookmarks/index.js [hunk 2]
```diff
@@ -45,8 +49,7 @@ module.exports = {
           }
           return -1;
         })
-        // return only top 20
-        .filter((a, b) => b < 20);
+        .slice(0, size)
```
Comment (adnasa) [inline resources/plugins/bookmarks/index.js]: there we go

Comment (vutran) [inline resources/plugins/bookmarks/index.js]: 👍

File: resources/plugins/bookmarks/index.js [hunk 3]
```diff
@@ -18,19 +18,23 @@ const mapDextItem = item => ({
...
+
+      // map to Dext items and apply scores
+      const items = bookmarks.map((bookmark) => {
+        const dextItem = mapDextItem(bookmark);
+        const score = dextItem.title.score(query);
+        return Object.assign({}, dextItem, {
+          score,
+        });
       });
+
```
Comment (adnasa) [inline resources/plugins/bookmarks/index.js]: 👆🏽clean up

**Comment by adnasa** [comment]
@vutran 7a9b2ed

**Pull request merged**